server/src/test/java/org/elasticsearch/action/support/GroupedActionListenerTests.java

```diff
@@ -133,4 +136,21 @@ public void testConcurrentFailures() throws InterruptedException {
         assertThat(exception, instanceOf(IOException.class));
         assertEquals(numGroups - 1, exception.getSuppressed().length);
     }
+
+    /*
+     * It can happen that the same exception causes a grouped listener to be notified of the failure multiple times. Since we suppress
+     * additional exceptions into the first exception, we have to guard against suppressing into the same exception, which could occur if we
+     * are notified of with the same failure multiple times. This test that the guard against self-suppression remains.
```

Comment:
NIT: seems like there's a word missing here like "this test [verifies] that the"

Reply:
Thanks. I pushed a commit.